File: java/src/main/java/problems1_100/MedianOfTwoSortedArrays.java
package problems1_100;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

enum MedianOfTwoSortedArrays {
    DEFAULT {
        @Override
        double findMedianSortedArrays(int[] nums1, int[] nums2) {
            List<Integer> nums = new ArrayList<>();
            for (int num : nums1) {
                nums.add(num);
            }
            for (int num : nums2) {
                nums.add(num);
            }
            Collections.sort(nums);
            int center = nums.size() / 2;
            return nums.size() % 2 == 1
                ? nums.get(center)
                : (double) (nums.get(center - 1) + nums.get(center)) / 2;
        }
    };

    abstract double findMedianSortedArrays(int[] nums1, int[] nums2);
}

File: kotlin/src/main/kotlin/problems1_100/MedianOfTwoSortedArrays.kt
package problems1_100

enum class MedianOfTwoSortedArrays {
    DEFAULT {
        override fun findMedianSortedArrays(nums1: IntArray, nums2: IntArray): Double {
            val nums = arrayListOf<Int>()
            for (num in nums1) {
                nums += num
            }
            for (num in nums2) {
                nums += num
            }
            nums.sort()
            val center = nums.size / 2
            return if (nums.size % 2 == 1) {
                nums[center].toDouble()
            } else {
                (nums[center - 1] + nums[center]).toDouble() / 2
            }
        }
    },
    ;

    abstract fun findMedianSortedArrays(nums1: IntArray, nums2: IntArray): Double
}
